### PR TITLE
interop/xds: dockerfile for the xds interop client

### PR DESCRIPTION
**Pull request by easwars**

### CODE REVIEW

**Comment by easwars** [comment]
@sergiitk : I'm not able to add you as a reviewer. cc'ing you here.

File: interop/xds/client/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# Build a static binary without cgo so that we can copy just the binary in the
+# final image, and can get rid of Go compiler and gRPC-Go dependencies.
+WORKDIR /go/src/grpc-go/interop/xds/client
+RUN go build -tags osusergo,netgo client.go
+
+# Second stage of the build which copies over only the client binary and skips
+# the Go compiler and gRPC repo from the earlier stage. This significantly
+# reduces the docker image size.
+FROM alpine
+COPY --from=build /go/src/grpc-go/interop/xds/client/client .
```
Comment (sergiitk) [inline interop/xds/client/Dockerfile:35]: If you adopting my previous suggestion, then it'll be
```suggestion
COPY --from=build /go/src/grpc-go/client .
```

**Review by sergiitk — COMMENTED** [review]
Looks good, minor comments/suggestions

File: interop/xds/client/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Copyright 2020 gRPC authors.
```
[the remark this answers is not in the document]
Comment (easwars) [reply interop/xds/client/Dockerfile:1]: Done.

File: interop/xds/client/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+# limitations under the License.
+
+# Dockerfile for building the xDS interop client. To build the image, run the
+# following command from grpc-go directory:
+# docker build -t <TAG> -f interop/xds/client/Dockerfile .
+
+FROM golang:1.16-alpine as build
+
+# Make a grpc-go directory and copy the repo into it.
+RUN mkdir /go/src/grpc-go
```
[the remark this answers is not in the document]
Comment (easwars) [reply interop/xds/client/Dockerfile:22]: Done.

File: interop/xds/client/Dockerfile (hunk 1 above)
Comment (easwars) [reply interop/xds/client/Dockerfile:35]: Done.

File: interop/xds/client/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+WORKDIR /go/src/grpc-go
+COPY . .
+
+# Build a static binary without cgo so that we can copy just the binary in the
+# final image, and can get rid of Go compiler and gRPC-Go dependencies.
+WORKDIR /go/src/grpc-go/interop/xds/client
+RUN go build -tags osusergo,netgo client.go
```
[the remark this answers is not in the document]
Comment (easwars) [reply interop/xds/client/Dockerfile:29]: Done.

**Review by easwars — COMMENTED** [review]
Thanks for your comments @sergiitk

**Review by sergiitk — APPROVED** [review]
👍 Thanks for the changes!

**Review by menghanl — APPROVED** [review]

**Pull request merged**